contracts/Prover.sol

```diff
@@ -342,26 +357,24 @@ contract Prover is SimpleProver {
         uint8 gameStatus,
         bool initialized,
         bool l2BlockNumberChallenged
-    ) public pure returns (bytes memory gameStatusStorageSlotRLP) {
-        // The if test is to remove leaing zeroes from the bytes
-        // Assumption is that initialized is always true
-        if (l2BlockNumberChallenged) {
-            gameStatusStorageSlotRLP = RLPWriter.writeBytes(
-                abi.encodePacked(l2BlockNumberChallenged, initialized, gameStatus, resolvedAt, createdAt)
-            );
-        } else {
-            gameStatusStorageSlotRLP = bytes.concat(
-                RLPWriter.writeBytes(
-                    abi.encodePacked(
-                        // abi.encodePacked(l2BlockNumberChallenged),
-                        initialized,
-                        gameStatus,
-                        resolvedAt,
-                        createdAt
+    ) public pure returns (bytes32 gameStatusStorageSlotRLP) {
+      // Packed data is 64 + 64 + 8 + 8 + 8 = 152 bits / 19 bytes.
+      // Need to convert to `uint152` to preserve right alignment.
+        return bytes32(
+            uint256(
+                uint152(
+                    bytes19(
+                        abi.encodePacked(
+                            l2BlockNumberChallenged,
```

Comment:
does this account for both of the cases you had before?

Reply:
Yes tested it with leading zero in the [AUDIT-RLP-TESTS](https://github.com/eco/eco-routes/tree/AUDIT-RLP-TESTS) branch SEE [here]([AUDIT-RLP-TESTS](https://github.com/eco/eco-routes/tree/AUDIT-RLP-TESTS)) and [here](https://github.com/eco/eco-routes/blob/AUDIT-RLP-TESTS/test/StorageRLP.spec.ts#L312-L415)